cpp/vector_kernels.hpp

```diff
@@ -0,0 +1,218 @@
+// Copyright (C) 2021 Sarah Roggendorf
+//
+// This file is part of DOLFINx_CUAS
+//
+// SPDX-License-Identifier:    LGPL-3.0-or-later
+
+#pragma once
+
+#include "kernels.hpp"
+#include "utils.hpp"
+#include <dolfinx/fem/FiniteElement.h>
+#include <dolfinx/fem/FunctionSpace.h>
+
+namespace dolfinx_cuas
+{
+
+kernel_fn generate_vector_kernel(dolfinx_cuas::Kernel type, int P)
+{
+  // Problem specific parameters
+  std::string family = "Lagrange";
```

Comment:
I would like this function to use the more general interface, as we did with the surface integrals for the matrix. 
I.e. this should support both triangles and tetrahedrons, with notes on how to extend it to non-affine geometries.